js/main.js

```diff
@@ -115,6 +117,171 @@ var makeMapActive = function () {
+    var adImgElement = adImg.cloneNode(true);
+    adImgElement.src = photo;
+    fragment.appendChild(adImgElement);
+  });
+  adPhotos.appendChild(fragment);
+};
+
+// Добавляет текст в элементы разметки объявления
+
+var addTextToElement = function (htmlElement, text) {
+  htmlElement.textContent = text;
+};
+
+// Добавляет цену в объявление
+
+var addPriceToAd = function (adPrice, price) {
+  adPrice.textContent = price + '₽/ночь';
+};
+
+// Добавляет тип жилья в объявление
+
+var addTypeToAd = function (adType, type) {
+  var typesMap = {
+    'flat': 'Квартира',
+    'bungalo': 'Бунгало',
+    'house': 'Дом',
+    'palace': 'Дворец'
+  };
+  adType.textContent = typesMap[type];
+};
+
+// Определяет склонение "комнат" в зависимости от числа
+
+var declineRooms = function (rooms) {
+  if ((rooms % 100 > 4 && rooms % 100 < 20) || rooms % 10 === 0 || rooms % 10 >= 5) {
+    return ROOMS_DECLENSIONS[2];
+  } else if (rooms % 10 < 5 && rooms % 10 > 1) {
+    return ROOMS_DECLENSIONS[1];
+  } else {
+    return ROOMS_DECLENSIONS[0];
+  }
+};
+
+// Определяет склонение "гостей" в зависимости от числа
+
+var declineGuests = function (guests) {
+  if (guests % 10 !== 1 || guests % 100 === 11) {
+    return GUESTS_DECLENSIONS[1];
+  } else {
+    return GUESTS_DECLENSIONS[0];
+  }
+};
+
+// Добавляет количество комнат и гостей в объявление
+
+var addCapacityToAd = function (adCapacity, rooms, guests) {
+  adCapacity.textContent = rooms + ' ' + declineRooms(rooms) + ' для ' + guests + ' ' + declineGuests(guests);
+};
+
+// Добавляет время въезда/выезда в объявление
+
+var addTimesToAd = function (adTiming, checkin, checkout) {
+  adTiming.textContent = 'Заезд после ' + checkin + ', выезд до ' + checkout;
+};
+
+// Добавляет аватар в объявление
+
+var addAvatarToAd = function (adAvatar, avatar) {
+  adAvatar.src = avatar;
+};
+
+// Добавляет удобства в объявление
+
+var addFeaturesToAd = function (adFeatures, features) {
+  var fragment = document.createDocumentFragment();
+  adFeatures.innerHTML = '';
+
+  features.forEach(function (feature) {
+    var clearItem = document.createElement('li');
+    var featureClass = 'popup__feature--' + feature;
+    clearItem.classList.add('popup__feature', featureClass);
+    fragment.appendChild(clearItem);
+  });
+
+  adFeatures.appendChild(fragment);
+};
+
+// Добавляет контент или скрывает элемент если контент пустой
+
+var addContentOrHide = function (htmlElement, callback, content, extraContent) {
+  if (Array.isArray(content)) {
+    if (content.length > 0) {
+      callback(htmlElement, content);
+    } else {
+      hideHtmlElement(htmlElement);
+    }
+  } else if (extraContent !== undefined) {
+    if (content && extraContent) {
+      callback(htmlElement, content, extraContent);
+    } else {
+      hideHtmlElement(htmlElement);
+    }
+  } else {
+    if (content) {
+      callback(htmlElement, content);
+    } else {
+      hideHtmlElement(htmlElement);
+    }
+  }
+};
```

Comment:
Это не очень. Давай сделаем так
```suggestion
var addNotEmptyContent = function (htmlElement, callback, content, extraContent) {
  if (callback(htmlElement, content, extraContent) === false) {
    hideHtmlElement(htmlElement);
  }
};
```
Я ее переименовал, чтоб не было `or` в названии.
Теперь тебе функции надо переделать так, чтоб они возвращаели ложь при пустом значении и истину при непустом. Так как у каждого своя логика, это нужно прописать в каждой функции. Например
```javascript
var addAvatarToAd = function (adAvatar, avatar) {
  if (!avatar) {
    return false;
  }
  adAvatar.src = avatar;
  return true;
};
```
Мне кажется, так более аккуратно выглядит

Reply:
Я бы ни за что не додумалась до такого, по крайней мере, на той стадии знания js, который у меня есть)) Круто! Спасибо!